pages/Network_APIs/CAMARA_QualityonDemand.md

```diff
@@ -14,16 +14,81 @@ This documentation is currently **under development and subject to change**. It
 
 Information: [https://github.com/camaraproject/QualityOnDemand](https://github.com/camaraproject/QualityOnDemand)
 
 The API definitions can be obtained here: [https://github.com/camaraproject/QualityOnDemand/tree/main/code/API_definitions](https://github.com/camaraproject/QualityOnDemand/tree/main/code/API_definitions)
 
+## Workflow: Media application requesting Quality on Demand
+
+A user of a media application would like to request the assignment of a QoS Profile to a device and be able to establish a QoS session for the connection between such a device and an application server. The following steps are executed:
+
+<figure>
+  <img src="./Content_Production/images/figure_qualityondemand.png" width="80%">
+</figure>
+
+### Step 0: Pre-conditions
+* qosProfiles have already been defined and made available by the network operator.
+* Names of such qosProfiles have been disclosed to the user so they can be used when invoking APIs.
+
+### Step 1: Check details of an existing QoS Profile
+* **GET /qos-profiles/{name}** to obtain the parameters of the QoS Profile
+
+### Step 2a: Attach a device to the QoS Profile
+* **POST /qos-assignments** passing the `device` object and the name of the `qosProfile`
+* The received "assignmentId" could be used to query details or revoke the assignment
+* This QoS profile will be assigned to the device anytime it is connected to the network.
+
+### Step 2b: Establish a QoS session
+* **POST /sessions** passing the `device` object, `applicationServer` IP, `applicationServerPorts`, `devicePorts`, `qosProfile` and `duration`.
+
+## 5G-MAG's Self-Assessment
+
+The QoS Profiles API would be used prior to the start of the event in order to understand the profiles and details of the profiles available in the network. However, it seems that the only way to obtain the list of parameters of a profile is by invoking the API with the QoS Profile `name`. There is a method lacking the query of profile names. There is also no method to define QoS Profiles, and this operation should be done beforehand.
+
+Before the event starts, device may be assigned to a given QoS Profile. This QoS Profile would be active for such device for the duration of the assignment or until it gets revoked. This would be useful for devices that require a static QoS Profile for the duration of the event. Impractical if the device would have to change as revoking the assignment may then result in not being available to assign a new device with a different QoS Profile (if there is contention with other users).
+
+A session may be created by establishing a guarantee of QoS between the device and the application server for a given duration. It is assumed that the QoS Provisioning would result successful when invoked in the given location. However as a service area cannot be defined/requested, it is unclear whether this would be successful or not.
```

Comment:
```suggestion
A session may be created by establishing a level of QoS between the device and the application server for a given duration. It is assumed that the QoS Provisioning would result successful when invoked in the given location. However as a service area cannot be defined/requested, it is unclear whether this would be successful or not.
```
Note, there can be different QoS Profiles, e.g. just different priority or others with an assured throughput.